TeamCode/src/main/java/org/firstinspires/ftc/teamcode/MecanumDrive.java

```diff
@@ -166,14 +177,17 @@ public Twist2dDual<Time> update() {
 
                 lastHeading = heading;
 
-                return new Twist2dDual<>(
+                delta = new Twist2dDual<>(
                         Vector2dDual.constant(new Vector2d(0.0, 0.0), 2),
                         DualNum.constant(0.0, 2)
                 );
+
+                pose = pose.plus(delta.value());
+                return delta.velocity().value();
```

Comment:
Applying a zero delta to `pose` shouldn't change it so you can just return zero velocity here.